api/v1/policyset_types.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// A custom type is required since there is no way to have a kubebuilder marker
+// apply to the items of a slice.
+
+// +kubebuilder:validation:MinLength=1
+type NonEmptyString string
+
+// PolicySetSpec defines the desired state of PolicySet
+type PolicySetSpec struct {
+	Description string `json:"description,omitempty"`
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Policies []NonEmptyString `json:"policies"`
```

Comment:
I think we should allow creating a policyset without adding any policies

Reply:
@ycao56 ok, I can change it back. Should I add `omitempty` or do we want an empty list to be required if the set has no policies?

Reply:
also, should I keep `NonEmptyString` or just make it a normal string type?

Reply:
`NonEmptyString` make sense.

Reply:
I think you can make it string and set `kubebuilder:validation:MinLength=1` on it

Reply:
oh, it won't work as it is a slice of strings not struct

Reply:
I think we want to keep it as `NonEmptyString`. I can't see a valid situation with an empty string.

Reply:
ok, i will just remove the MinItems=1 and leave the nonemptystring then